ethcore/src/engines/mod.rs

```diff
@@ -71,8 +71,9 @@ pub trait Engine : Sync + Send {
 	/// Block transformation functions, after the transactions.
 	fn on_close_block(&self, _block: &mut ExecutedBlock) {}
 
-	/// If true, generate_seal has to be implemented.
-	fn seals_internally(&self) -> bool { false }
+	/// If Some(true) this author is able to generate seals, generate_seal has to be implemented.
```

Comment:
`Option<bool>` is convenient for `unwrap_or` or `map_or` consumers but a custom enum would express intent better. I would chalk this one up to personal style -- as long as the docs explain what the `Option` signifies here it should be ok.